### PR TITLE
Enable FusedRoPE using float32 for gpt-neox model

### PR DESCRIPTION
**Pull request by yeonsily**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by mandy-li — CHANGES_REQUESTED** [review]

File: optimum/habana/transformers/models/gpt_neox/modeling_gpt_neox.py [hunk 1]
```diff
@@ -422,24 +417,27 @@ def gaudi_gpt_neox_rotary_embedding_set_cos_sin_cache(self, seq_len, device, dty
...
-        if k.dtype == torch.bfloat16:
-            rope_k = FusedRoPE.apply(
-                k,
-                cos.unsqueeze(0).unsqueeze(0).to(torch.bfloat16),
-                sin.unsqueeze(0).unsqueeze(0).to(torch.bfloat16),
-                position_ids,
-            )
-        else:
-            rope_k = FusedRoPE.apply(k, cos.unsqueeze(0).unsqueeze(0), sin.unsqueeze(0).unsqueeze(0), position_ids)
+        # TODO: Enable FusedRoPE with bf16 after accuracy is fixed
```
[the remark this answers is not in the document]
Comment (yeonsily) [reply optimum/habana/transformers/models/gpt_neox/modeling_gpt_neox.py:420]: oh, it's my mistake. let me fix it.

File: optimum/habana/transformers/models/gpt_neox/modeling_gpt_neox.py [hunk 2]
```diff
@@ -415,29 +415,30 @@ def gaudi_gpt_neox_rotary_embedding_set_cos_sin_cache(self, seq_len, device, dty
...
-        #             k,
-        #             cos.unsqueeze(0).unsqueeze(0).to(torch.bfloat16),
-        #             sin.unsqueeze(0).unsqueeze(0).to(torch.bfloat16),
-        #             position_ids,
-        #         )
-        #     else:
-        #         rope_k = FusedRoPE.apply(k, cos.unsqueeze(0).unsqueeze(0), sin.unsqueeze(0).unsqueeze(0), position_ids)
-        rope_q = FusedRoPE.apply(q, cos.unsqueeze(0).unsqueeze(0), sin.unsqueeze(0).unsqueeze(0), position_ids)
-        rope_k = FusedRoPE.apply(k, cos.unsqueeze(0).unsqueeze(0), sin.unsqueeze(0).unsqueeze(0), position_ids)
+        if training:  # TODO: Enable FusedRoPE with bf16 after accuracy is fixed
```
Comment (mandy-li) [inline optimum/habana/transformers/models/gpt_neox/modeling_gpt_neox.py:420]: You may remove #TODO since this is to use the fix for accuracy issue

**Review by mandy-li — COMMENTED** [review]
LGTM except the minor comment

File: optimum/habana/transformers/models/gpt_neox/modeling_gpt_neox.py (hunk 2 above)
Comment (yeonsily) [reply optimum/habana/transformers/models/gpt_neox/modeling_gpt_neox.py]: Done.

**Review by mandy-li — APPROVED** [review]

**Review by regisss — APPROVED** [review]

**Pull request merged**